chainer/function_node.py

```diff
@@ -1204,13 +1205,31 @@ def _extract_apply_in_data(inputs):
                     has_chainerx_array = True
                 else:
                     arrays.append(x.array)
+                    nonchainerx_requires_grad |= x.requires_grad
             else:  # x is ndarray
                 arrays.append(x)
                 if not has_chainerx_array:
                     if isinstance(x, chainerx.ndarray):
                         has_chainerx_array = True
 
         if has_chainerx_array:
+            if nonchainerx_requires_grad and chainer.config.enable_backprop:
+                msg = []
+                msg.append(
+                    'Cannot mix ChainerX and non-ChainerX inputs because '
+                    'ChainerX\'s backprop does not support Variables on '
+                    'other backends')
+                for i, x in enumerate(inputs):
+                    typ = type(chainer.Variable()).__name__
+                    if isinstance(x, variable.Variable):
+                        typ += '(requires_grad={})'.format(x.requires_grad)
+                        device = x.device
+                    else:
+                        device = backend.get_device_from_array(x)
+                    msg.append(
+                        'input[{}]: {} on {}'.format(
+                            i, typ, device))
+                raise NotImplementedError('\n'.join(msg))
```

Comment:
How about `RuntimeError`? I think it's highly unlikely it will ever be supported.

Reply:
Changed it to `RuntimeError`.